src/app/redux/AppReducer.js

```diff
@@ -24,55 +34,93 @@ const defaultState = Map({
-    }
-    if (action.type === 'FETCH_DATA_END') {
-        res = state.set('loading', false);
-    }
-    if (action.type === 'ADD_NOTIFICATION') {
-        const n = {
-            action: tt('g.dismiss'),
-            dismissAfter: 10000,
-            ...action.payload
-        };
-        res = res.update('notifications', s => {
-            return s ? s.set(n.key, n) : OrderedMap({[n.key]: n});
-        });
-    }
-    if (action.type === 'REMOVE_NOTIFICATION') {
-        res = res.update('notifications', s => s.delete(action.payload.key));
-    }
-    if (action.type === 'UPDATE_NOTIFICOUNTERS' && action.payload) {
-        const nc = action.payload;
-        if (nc.follow > 0) {
-            nc.total -= nc.follow;
-            nc.follow = 0;
+    switch (action.type) {
+        case '@@router/LOCATION_CHANGE':
+            return state.set('location', { pathname: action.payload.pathname });
+        case STEEM_API_ERROR:
+            return  state.set('error', action.error).set('loading', false);
+        case FETCH_DATA_BEGIN:
+            return state.set('loading', true);
+        case FETCH_DATA_END:
+            return state.set('loading', false);
+        case ADD_NOTIFICATION: {
+            const n = {
+                action: tt('g.dismiss'),
+                dismissAfter: 10000,
+                ...action.payload
+            };
+            return state.update('notifications', (s) => {
+                return s ? s.set(n.key, n) : OrderedMap({[n.key]: n});
+            });
         }
-        res = res.set('notificounters', Map(nc));
-    }
-    if (action.type === 'SET_USER_PREFERENCES') {
-        res = res.set('user_preferences', Map(action.payload));
-    }
-    if (action.type === 'TOGGLE_NIGHTMODE') {
-        res = res.setIn(['user_preferences', 'nightmode'], !res.getIn(['user_preferences', 'nightmode']));
-    }
-    if (action.type === 'TOGGLE_BLOGMODE') {
-        res = res.setIn(['user_preferences', 'blogmode'], !res.getIn(['user_preferences', 'blogmode']));
+        case REMOVE_NOTIFICATION:
+            return state.update('notifications', s => s.delete(action.payload.key));
+        case UPDATE_NOTIFICOUNTERS: {
+            if (action.payload) {
+                const nc = action.payload;
+                if (nc.follow > 0) {
+                    nc.total -= nc.follow;
+                    nc.follow = 0;
+                }
+                return state.set('notificounters', Map(nc));
+            }
+            return state;
+        }
+        case SET_USER_PREFERENCES:
+            return state.set('user_preferences', Map(action.payload));
+        case TOGGLE_NIGHTMODE:
+            return state.setIn(['user_preferences', 'nightmode'], !state.getIn(['user_preferences', 'nightmode']));
+        case TOGGLE_BLOGMODE:
+            return state.setIn(['user_preferences', 'blogmode'], !state.getIn(['user_preferences', 'blogmode']));
+        default:
+            return state;
     }
-    return res;
 }
```

Comment:
The `notifications-ui` branch does exactly this as well. You probably know that. We'll need to keep tabs when merging it in.

Reply:
not in this reducer though -- we're only working in new reducers so it'll be an easy merge